seclea_ai/internal/file_processor.py

```diff
@@ -184,8 +516,16 @@ def _save_model_state(
                 training_run=str(training_run_pk),
             )
             self._dbms.save_datasetmodelstate(ds, "stored")
-
-            onSuccess(save_path, training_run_pk, sequence_num, final)
+            # pushing data to queue
+            self._sender_q.put(
+                {
+                    "function": "_send_model_state",
+                    "save_path": save_path,
+                    "training_run_pk": training_run_pk,
+                    "sequence_num": sequence_num,
+                    "final": final,
+                }
+            )
```

Comment:
This shouldn't be here - add to interface.

Reply:
If we put outside then we need to use callable function to call this otherwise we won't get this data which we need to send to server

Reply:
Not 100% what you mean by that, do you mean that we won’t have the save_path if we don’t call it from here?

Reply:
Yes save_path is generating in this function only that's why we have called from here

Reply:
I have refactored, resolving this.